File: ExampleiOS/ExampleiOSTests/TMBaseTestCase.m
//
//  TMBaseTestCase.m
//  ExampleiOSTests
//
//  Created by Pinar Olguc on 10.05.2021.
//  Copyright © 2021 tumblr. All rights reserved.
//

#import "TMBaseTestCase.h"

@implementation TMBaseTestCase

- (void)setUp {
    [super setUp];
    [self createTempDirectory];
}

- (void)tearDown {
    [self removeAllItemsInsideTempDirectory];
}

//MARK: - Temp directory utils

- (NSURL *)tempTestDirectory {
    return [[[NSFileManager defaultManager] temporaryDirectory] URLByAppendingPathComponent:@"com.tumblr.sdk"];
}

- (NSURL *)tempFileURL {
    return [[self tempTestDirectory] URLByAppendingPathComponent: [[NSUUID UUID] UUIDString]];
}

- (BOOL)removeAllItemsInsideTempDirectory
{
    __block BOOL result = YES;
    NSDirectoryEnumerator<NSString *> *enumerator = [[NSFileManager defaultManager] enumeratorAtPath:[self tempTestDirectory].path];
    NSString *fileName;
    while (fileName = [enumerator nextObject])
    {
        NSURL *fileURL = [[self tempTestDirectory] URLByAppendingPathComponent: fileName];
        BOOL success = [[NSFileManager defaultManager] removeItemAtPath:fileURL.path error:nil];
        if (!success) {
            result = NO;
        }
    }
    return result;
}

- (BOOL)createTempDirectory
{
    NSError *error;
    [[NSFileManager defaultManager] createDirectoryAtURL:self.tempTestDirectory withIntermediateDirectories:YES attributes:nil error:&error];
    return error == nil;
}

@end

File: ExampleiOS/ExampleiOSTestsSwift/TMBaseTestCase.swift
//
//  File.swift
//  
//
//  Created by Adriana Elizondo on 01/04/24.
//

import Foundation
import XCTest

class TMBaseTestCase: XCTestCase {
    
    override func setUp() {
        super.setUp()
        createTempDirectory()
    }
    
    override func tearDown() {
        removeAllItemsInsideTempDirectory()
        super.tearDown()
    }
    
    // MARK: - Temp directory utils
    func tempTestDirectory() -> URL {
        return FileManager.default.temporaryDirectory.appendingPathComponent("com.tumblr.sdk")
    }
    
    func tempFileURL() -> URL {
        return tempTestDirectory().appendingPathComponent(UUID().uuidString)
    }
    
    @discardableResult
    func removeAllItemsInsideTempDirectory() -> Bool {
        let fileManager = FileManager.default
        let enumerator = fileManager.enumerator(atPath: tempTestDirectory().path)
        var result = true
        
        while let fileName = enumerator?.nextObject() as? String {
            let fileURL = tempTestDirectory().appendingPathComponent(fileName)
            do {
                try fileManager.removeItem(at: fileURL)
            } catch {
                result = false
            }
        }
        
        return result
    }
    
    @discardableResult
    func createTempDirectory() -> Bool {
        do {
            try FileManager.default.createDirectory(at: tempTestDirectory(), withIntermediateDirectories: true, attributes: nil)
            return true
        } catch {
            return false
        }
    }
}
